File: leetcode/1493 Longest Subarray of 1's After Deleting One Element/solution.go
func longestSubarray(nums []int) int { // 滑动窗口虫取法！！！
    k, zero := 1, 0
    left, right := 0, 0
    for right < len(nums) {
        if nums[right] == 0 {
            zero++
        }
        if zero > k {
            if (nums[left] == 0) {
                zero--
            }
            left++
        }
        right++
    }
    return right-left-1
}

File: leetcode/1493 Longest Subarray of 1's After Deleting One Element/solution.java
class Solution {
    public int longestSubarray(int[] nums) { // 滑动窗口虫取法！！！
        int k = 1;
        int zero = 0;
        int left = 0, right = 0;
        while (right < nums.length) {
            if (nums[right] == 0) {
                zero++;
            }
            if (zero > k) {
                if (nums[left] == 0) zero--;
                left++; 
            }
            right++;
        }
        return right-left-1;
    }
}